### PR TITLE
S19

### PR DESCRIPTION
**Pull request by devwistkey**

### CODE REVIEW

File: components/Layout/index.tsx [hunk 1]
```diff
@@ -66,13 +60,28 @@ function LayoutSearch(props: LayoutProps) {
...
 
     const bg = loginInfo.isLight ? 'bglight' : 'bg'
 
     const checkZilpayNetwork = async () => {
         if (loginInfo.zilAddr) {
-            updateShowZilpay(true)
+            const wallet = new ZilPayBase()
+            const zp = await wallet.zilpay()
+            const network = zp.wallet.net
+            if (network !== loginInfo.net) {
```
Comment (tralkan) [inline components/Layout/index.tsx:71]: hey we need to verify the ZilPay wallet change even if the network remains the same @devwistkey

File: components/Layout/index.tsx [hunk 2]
```diff
@@ -133,7 +142,6 @@ function LayoutSearch(props: LayoutProps) {
                         </>
                     )}
                 <Footer />
-                {showZilpay && <ZilPay />}
```
Comment (tralkan) [inline components/Layout/index.tsx:136]: Then where are we running this component? We could call the needed function instead.

File: components/Footer/index.tsx [hunk 3]
```diff
@@ -139,11 +139,11 @@ function Footer() {
                         // @info why the router here does not work? URL update but UI not: because when we're pushing to the
                         // same page e.g /ilhamb to /ssiprotocol it'll not trigger useeffect (but if from ilhamb/didx to /ssiprotocol this is works)
                         // Router.push('/ssiprotocol/tree')
-                        window.open('http://tyron.network/ssiprotocol', '_self')
+                        // window.open('http://tyron.network/ssiprotocol', '_self')
```
Comment (tralkan) [inline components/Footer/index.tsx:142]: why commented out?

File: components/Dashboard/index.tsx [hunk 4]
```diff
@@ -60,11 +58,11 @@ function Component() {
...
     }
 
-    // useEffect(() => {
-    //     if (loginInfo.zilAddr !== null) {
-    //         updateShowZilpay(false)
-    //     }
-    // }, [loginInfo.zilAddr])
+    useEffect(() => {
+        if (loginInfo.zilAddr !== null) {
+            updateShowZilpay(false)
```
Comment (tralkan) [inline components/Dashboard/index.tsx:63]: this is not handling the wallet switch properly

**Review by tralkan — CHANGES_REQUESTED** [review]

File: components/Modals/DashboardModal/index.tsx [hunk 5]
```diff
@@ -283,15 +284,19 @@ function Component() {
                                     `https://v2.viewblock.io/zilliqa/tx/${deploy[0].ID}?network=${net}`
                                 )
                             }, 1000)
-                            let new_ssi = deploy[0].ContractAddress
-                            new_ssi = zcrypto.toChecksumAddress(new_ssi)
+                            let api = 'https://api.zilliqa.com'
+                            if (net === 'mainnet') {
+                                api = 'https://dev-api.zilliqa.com'
```
Comment (tralkan) [inline components/Modals/DashboardModal/index.tsx:289]: Hey @devwistkey, it's the other way around, dev for testnet

File: components/Modals/DashboardModal/index.tsx [hunk 6]
```diff
@@ -283,15 +284,19 @@ function Component() {
...
+                            }
+                            const zilliqa = new Zilliqa(api)
+                            const txn =
+                                await zilliqa.blockchain.getContractAddressFromTransactionID(
+                                    deploy[0].ID
+                                )
+                            const new_ssi = '0x' + txn.result
```
Comment (tralkan) [inline components/Modals/DashboardModal/index.tsx:296]: Remember to always normalise addresses. Also test is buy nft in this case works (have to test from ZilPay mobile @devwistkey @koniandriani)

**Review by tralkan — CHANGES_REQUESTED** [review]
Pls address the comments.

File: components/Modals/BuyNFT/index.tsx [hunk 7]
```diff
@@ -68,6 +71,31 @@ function Component() {
     const [inputAddr, setInputAddr] = useState('')
     const [legend, setLegend] = useState('save')
     const [loading, setLoading] = useState(false)
+    const [loadingPayment, setLoadingPayment] = useState(false)
+
+    // const submitAr = async () => {
+    //     try {
+    //         const arweave = Arweave.init({
```
Comment (tralkan) [inline components/Modals/BuyNFT/index.tsx:78]: Use arweave from constants file

**Review by tralkan — CHANGES_REQUESTED** [review]

File: components/Layout/index.tsx (hunk 1 above)
Comment (devwistkey) [reply components/Layout/index.tsx:71]: Hi yes, you're right. I just update and now wallet switch works

File: components/Layout/index.tsx (hunk 2 above)
Comment (devwistkey) [reply components/Layout/index.tsx:136]: We already show the component on Dashboard.tsx, if we call again it'll causing duplication and infinity loop

File: components/Footer/index.tsx (hunk 3 above)
Comment (devwistkey) [reply components/Footer/index.tsx:142]: Last time I disable this so Koni can do testing and inspect the resolved addr. Fixed, I just put it back

File: components/Dashboard/index.tsx (hunk 4 above)
Comment (devwistkey) [reply components/Dashboard/index.tsx:63]: Hi @tralkan wallet switch working now after removing network condition(if network !== loginInfo.net). I just post comment on notion

File: components/Modals/DashboardModal/index.tsx (hunk 5 above)
Comment (devwistkey) [reply components/Modals/DashboardModal/index.tsx:289]: sorry, fixed

File: components/Modals/BuyNFT/index.tsx (hunk 7 above)
Comment (devwistkey) [reply components/Modals/BuyNFT/index.tsx:78]: done

File: components/ZilPay/index.tsx [hunk 8]
```diff
@@ -62,7 +68,20 @@ export const ZilPay: React.FC = () => {
...
-                        //@todo-i if logged in & address is not the DID Controller, then log off
+                        if (loginInfo.address) {
+                            await tyron.SearchBarUtil.default
+                                .Resolve(loginInfo.net, loginInfo.address)
+                                .then(async (result: any) => {
+                                    const did_controller =
+                                        zcrypto.toChecksumAddress(
+                                            result.controller
+                                        )
+                                    if (did_controller !== address?.base16) {
```
Comment (tralkan) [inline components/ZilPay/index.tsx:79]: remember to normalise addresses

**Review by tralkan — APPROVED** [review]

File: components/Dashboard/index.tsx (hunk 4 above)
Comment (tralkan) [reply components/Dashboard/index.tsx:63]: what if loginInfo.zilAddr != null but still a switch? not updating here

File: components/ZilPay/index.tsx (hunk 8 above)
Comment (tralkan) [reply components/ZilPay/index.tsx:79]: I lowercased it.

**Pull request merged**